### PR TITLE
Updated fitting and adding LinKK validation method

### PR DESCRIPTION
**Pull request by mdmurbach**
- Adds `validation.linKK` method which implements the validation algorithm from [Schönleber et al.](https://dx.doi.org/10.1016/j.electacta.2014.01.034) and created `examples/validation_example.ipynb` with an example of how to use

- Cleans up the fitting module to use `scipy.optimize.curve_fit` which accepts the same methods, bounds, etc. as in the previous impedance.py implementation, but in a much more concise and readable way.

- Improves bootstrapping methods for confidence band calculation and visualization

### CODE REVIEW

File: docs/source/examples/validation_example.ipynb [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "import sys\n",
+    "sys.path.append(\"/Users/matt/Dropbox/UW-Files/Research/impedance.py/\")\n",
```
Comment (nealde) [inline docs/source/examples/validation_example.ipynb]: Hard-coded path to sys for loading in the package

suggested change: sys.path.append("../../../")

Comment (mdmurbach) [reply docs/source/examples/validation_example.ipynb]: Good catch 👍

Comment (nealde) [reply docs/source/examples/validation_example.ipynb]: There's another instance further down

**Comment by mdmurbach** [comment]
Thanks a bunch, Neal! I'll push those changes tonight.

Also working on finishing up a rewritten circuit algorithm that allows more
flexible (i.e. nested parallel groups) circuts.

Have a good Monday!

**Comment by mdmurbach** [comment]
Should be good to go now, thanks Neal!

**Comment by nealde** [comment]
No problem!  👍 

And oh man, good luck!

**Comment by mdmurbach** [comment]
@all-contributors please add @nealde for review

**Pull request merged**